src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double loadFactor = 0.75;
+    private int size;
+    private int initialCapacity = 16;
+    private Node<K, V>[] nodeArray;
+
+    public MyHashMap() {
+        nodeArray = new Node[initialCapacity];
+    }
+
+    class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node node = new Node();
+        if (key == null) {
+            node.hash = 0;
+        } else {
+            node.hash = Math.abs(key.hashCode()) % initialCapacity;
+        }
```

Comment:
lets move logic of finding backet index by hash to a separate method (as we use it several times in code)